### PR TITLE
admin can:

### PR DESCRIPTION
**Pull request by DemchyshynV**
+create Courses
+edit Courses
+delete Courses
+add/delete modules to course

### CODE REVIEW

File: src/validators/module/moduleValidator.ts [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
-export const moduleValidator  = Joi.object().keys({
-  label: Joi.string().max(255).trim().required(),
-  description: Joi.string().max(255).trim(),
-  tags: Joi.array().items(Joi.string()),
-  courses_id: Joi.array().items(Joi.string()),
-  lessons: Joi.array().items(Joi.string())
+import {lessonValidator} from "../lesson";
+
+export const moduleValidator = Joi.object().keys({
+    _id: Joi.string().max(255).trim(),
```
Comment (VictorKmin) [inline src/validators/module/moduleValidator.ts]: fix

File: src/services/module/module.service.ts [hunk 2]
```diff
@@ -12,37 +12,41 @@ class ModuleService {
...
       .populate('lesson_list')
-      .select({ _id: 0 }) as any;
+      .select({_id: 0}) as any;
   }
 
   editModule(module_id: string, updating_value: Partial<IModule>): Promise<void> {
     const ModuleModel = model<ModuleType>(DatabaseTablesEnum.MODULE_COLLECTION_NAME);
 
-    return ModuleModel.updateOne({ module_id }, { updating_value }) as any;
+    return ModuleModel.updateOne({module_id}, {updating_value}) as any;
```
[the remark this answers is not in the document]
Comment (DemchyshynV) [reply src/services/module/module.service.ts]: it`s not mine

File: src/services/user/user.service.ts [hunk 3]
```diff
@@ -1,57 +1,57 @@
...
+  }
+
+  getUserByParams(params: Partial<IUser>): Promise<any> {
+    const UserModel = model<UserType>(DatabaseTablesEnum.USER_COLLECTION_NAME, UserSchema);
+    return UserModel.findOne(params).select('+password') as any;
+  }
+
+  changeStatus(user_id: string, status: number): Promise<any> {
+    const UserModel = model<UserType>(DatabaseTablesEnum.USER_COLLECTION_NAME, UserSchema);
+    return UserModel.findByIdAndUpdate(user_id, {status_id: status}) as any;
```
[the remark this answers is not in the document]
Comment (DemchyshynV) [reply src/services/user/user.service.ts]: {status_id} - ok
{new: true} I do not make the second request, but only change Front

File: src/services/course/course.service.ts [hunk 4]
```diff
@@ -2,51 +2,63 @@ import { model } from 'mongoose';
...
+      .populate('modules_list')
+      .limit(limit)
+      .skip(skip)
+      .sort(order) as any;
+  }
+
+  getSizeOfAll(filterParams: Partial<ICourse>): Promise<number> {
+    const GroupModel = model<CourseType>(DatabaseTablesEnum.COURSE_COLLECTION_NAME, CourseSchema);
+    return GroupModel
+      .countDocuments({...filterParams}) as any;
```
Comment (VictorKmin) [inline src/services/course/course.service.ts]: Again?

File: src/services/course/course.service.ts [hunk 5]
```diff
@@ -2,51 +2,63 @@ import { model } from 'mongoose';
...
+    return CourseModel
+      .findOne({_id: course_id})
+      .populate('modules_list')
+      .select({label: 1, module_list: 1, _id: 0}) as any;
+  }
+
+  getCourses(filterParams: Partial<ICourse>, limit: number, skip: number, order: string): Promise<ICourse[]> {
+    const CourseModel = model<CourseType>(DatabaseTablesEnum.COURSE_COLLECTION_NAME, CourseSchema);
+    return CourseModel
+      .find({...filterParams})
```
Comment (VictorKmin) [inline src/services/course/course.service.ts]: .find(filter) ->   .find({...filterParams})
REALY ?

**Pull request merged**